From 82f7b78425f81f8e54b6c7605df507ade8f625dd Mon Sep 17 00:00:00 2001
From: Radu-Andrei BULIE <radu.bulie@freescale.com>
Date: Tue, 30 Oct 2012 00:01:55 +0000
Subject: [PATCH 360/518] dpa_classifier: Fix classifier compat header manip
 functions insert and vlan

Added fixes for insert and vlan manip functions when retrieving operation
type from userspace

Signed-off-by: Radu-Andrei BULIE <radu.bulie@freescale.com>
[Grabbed from the branch, LINUX_IR5.2.0, of
https://git.freescale.com/git-private/cgit.cgi/ppc/alu-b4860/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 .../staging/fsl_dpa_offload/wrp_dpa_classifier.c   |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/wrp_dpa_classifier.c b/drivers/staging/fsl_dpa_offload/wrp_dpa_classifier.c
index f3bed15..317c751 100644
--- a/drivers/staging/fsl_dpa_offload/wrp_dpa_classifier.c
+++ b/drivers/staging/fsl_dpa_offload/wrp_dpa_classifier.c
@@ -2369,6 +2369,8 @@ int dpa_cls_hm_insert_params_compatcpy(
 {
 	int type;
 
+	kparam->ins_params.type = uparam->ins_params.type;
+
 	type = kparam->ins_params.type;
 	switch (type) {
 	case DPA_CLS_HM_INSERT_CUSTOM:
@@ -2416,6 +2418,7 @@ int dpa_cls_hm_vlan_params_compatcpy(
 {
 	int type;
 
+	kparam->vlan_params.type = uparam->vlan_params.type;
 	type = kparam->vlan_params.type;
 	switch (type) {
 	case DPA_CLS_HM_VLAN_INGRESS:
-- 
1.7.5.4

